### PR TITLE
Task/dtuw 70 digitax re captcha v3 integration

### PR DESCRIPTION
**Pull request by Bithika-Redapple**
### What Does This PR Do?
recaptcha v3 integration

### Where Should The Reviewer Start?
signup.component.ts
signup.component.html
auth.service.ts

### Checklist (if applicable):
[ ] item 1
[ ] item 2
[ ] item 3
[ ] item 4
[ ] item 5

### CODE REVIEW

File: src/app/auth/auth.service.ts [hunk 1]
```diff
@@ -179,4 +200,45 @@ var ProductName = sessionStorage.getItem('ProductName');
...
+
+
+
+
+let data={
+//6LeJqqsZAAAAADgHTJGWg4YMI028ffU6c--Av2YR(client)
+//6LdoiLwZAAAAADmQpBrZnki6eFbWJS-WaD1r1luU---owner
+//secret:"6LdoiLwZAAAAADmQpBrZnki6eFbWJS-WaD1r1luU",
+
+   secret:"6LdoiLwZAAAAADmQpBrZnki6eFbWJS-WaD1r1luU",
```
Comment (drxeno02) [inline src/app/auth/auth.service.ts:235]: @Bithika-Redapple Do not hardcode the secret here. That is a security risk. Create a Constants class. Do not use string literals. Even better would be to have you keys in a property file.

File: src/app/auth/auth.service.ts [hunk 2]
```diff
@@ -179,4 +200,45 @@ var ProductName = sessionStorage.getItem('ProductName');
...
+   response:token
+}
+
+      const body = new HttpParams({fromObject: data});
+    const options = { headers: this.headers1};
+    return this.http.post(`https://www.google.com/recaptcha/api/siteverify`, body.toString(), options);
```
Comment (drxeno02) [inline src/app/auth/auth.service.ts:241]: @Bithika-Redapple Do not hardcode this url value. Retrieve the url from a Constants

File: src/app/auth/auth.service.ts (hunk 1 above)
Comment (Bithika-Redapple) [reply src/app/auth/auth.service.ts:235]: ok

File: src/app/home/home.component.ts [hunk 3]
```diff
@@ -35,6 +35,9 @@ export class HomeComponent implements OnInit {
         console.log(this.loginForm.value);
          sessionStorage.setItem('access_token', '234');
               sessionStorage.setItem('user_id', '123');
+
+              localStorage.setItem('access_token', '234');
```
Comment (drxeno02) [inline src/app/home/home.component.ts:39]: @Bithika-Redapple These keys should all be constants. You want one point of reference for these key values, which will then allow you to reuse. Update your code structure and architect to not have to hardcode any values, anywhere. Never hardcode.

File: src/app/home/home.component.ts [hunk 4]
```diff
@@ -35,6 +35,9 @@ export class HomeComponent implements OnInit {
...
+              localStorage.setItem('user_id', '123');
               let isLoggedIn = (localStorage.getItem('access_token'))?true:false; 
               //this.isLoggedIn = true;
```
Comment (drxeno02) [inline src/app/home/home.component.ts:42]: @Bithika-Redapple Remove deadcode. There is no reason to have this here.

File: src/app/signup/signup.component.ts [hunk 5]
```diff
@@ -24,72 +26,112 @@ export class SignupComponent implements OnInit {
...
+      this.isSubmitted = true;         
+      
+      
       if(this.signupForm.valid)
       {
-      let saveData = this.signupForm.value;   
-       
+       let saveData = this.signupForm.value;   
+          grecaptcha.ready(() => {
+              grecaptcha.execute('6LdoiLwZAAAAANJ-MV-ZORWzs8IwU1IjDPJcXnvn', { action: 'cta_signup' }).then((token) => {
```
Comment (drxeno02) [inline src/app/signup/signup.component.ts:93]: @Bithika-Redapple The action can be a Constant, and as mentioned before the secret can be a constant someplace. Hopefully in  some property or configuration file where it is less obvious where it is at.

File: src/app/signup/signup.component.ts [hunk 6]
```diff
@@ -24,72 +26,112 @@ export class SignupComponent implements OnInit {
...
+                       
+                      }
+                      else if(signupres['status'].status_code == 400){
+                         this.snackbar.open(signupres['status']['message'],'OK',{
+                          verticalPosition: 'top',
+                          horizontalPosition:'right'
+                        });
+                      }
+                       else{                      
+                       confirm('Sorry, an error occurred. Please email support@digitaltaxusa.com');
```
Comment (drxeno02) [inline src/app/signup/signup.component.ts:118]: @Bithika-Redapple You are using this same error message multiple times. Create a Constant out of it and have one point of reference.

File: src/environments/environment.ts [hunk 7]
```diff
@@ -4,7 +4,9 @@
 
 export const environment = {
   production: false,
-  BASE_URL   :"http://52.20.93.40/"
+  
+  BASE_URL:`http://digitaxapi-env.eba-nrr834zb.us-east-1.elasticbeanstalk.com:8080`
+  //CAPTCHA_URL   :`https://www.google.com/recaptcha/api/siteverify`
```
Comment (drxeno02) [inline src/environments/environment.ts:9]: @Bithika-Redapple Remove commented out code. Keep all codebase clean.

File: src/app/auth/auth.service.ts [hunk 8]
```diff
@@ -228,17 +216,16 @@ var ProductName = sessionStorage.getItem('ProductName');
...
 }
 
       const body = new HttpParams({fromObject: data});
-    const options = { headers: this.headers1};
-    return this.http.post(`https://www.google.com/recaptcha/api/siteverify`, body.toString(), options);
+    const options = { headers: this.headers1}; 
+
+    
+
+      return this.http.post(`${environment.CAPTCHA_URL}`, body.toString(), options);
```
Comment (drxeno02) [inline src/app/auth/auth.service.ts:228]: 👍

File: src/app/home/home.component.ts [hunk 9]
```diff
@@ -32,21 +36,36 @@ export class HomeComponent implements OnInit {
...
-              location.href = '/tax-prepare-profile'; 
-             // location.href = '/dashboard'; 
+        console.log(this.loginForm.value);         
+        
+        this.authService.login(this.f.email.value, this.f.password.value, this.f.remember_me.value,)
+            .pipe(first())
+            .subscribe(
+                resp => {
+                console.log("###");
+                console.log(resp);
```
Comment (drxeno02) [inline src/app/home/home.component.ts:46]: @Bithika-Redapple Stay away for any short hand. For example, do not say "resp", say "response". If you can add descriptor that is even better. For example, `captchaResponse`. I am noticing a lot of shorthand which is not clear and can be misinterpreted.

File: src/app/signup/signup.component.ts [hunk 10]
```diff
@@ -90,8 +93,8 @@ export class SignupComponent implements OnInit {
       {
        let saveData = this.signupForm.value;   
           grecaptcha.ready(() => {
-              grecaptcha.execute('6LdoiLwZAAAAANJ-MV-ZORWzs8IwU1IjDPJcXnvn', { action: 'cta_signup' }).then((token) => {
-                   console.log(token);
+              grecaptcha.execute(environment.site, { action: 'cta_signup' }).then((token) => {
```
Comment (drxeno02) [inline src/app/signup/signup.component.ts:96]: @Bithika-Redapple You could have named this as `site_key` to be more clear

File: src/app/signup/signup.component.ts [hunk 11]
```diff
@@ -114,15 +117,16 @@ export class SignupComponent implements OnInit {
...
+
+                       confirm(environment.error_message);
                        
                     } 
                  });
                       }
                        else{
                           /*show error message as recaptcha thresold value not ok*/
-                          confirm('Sorry, an error occurred. Please email support@digitaltaxusa.com');
+                          confirm(environment.error_message);
```
Comment (drxeno02) [inline src/app/signup/signup.component.ts:129]: @Bithika-Redapple Instead off error_message as I am sure there will be more error messages. You could have labeled this as `default_error_message`. Something like that.

File: src/environments/environment.ts [hunk 12]
```diff
@@ -5,8 +5,12 @@
 export const environment = {
   production: false,
   
-  BASE_URL:`http://digitaxapi-env.eba-nrr834zb.us-east-1.elasticbeanstalk.com:8080`
-  //CAPTCHA_URL   :`https://www.google.com/recaptcha/api/siteverify`
+  BASE_URL:`http://digitaxapi-env.eba-nrr834zb.us-east-1.elasticbeanstalk.com:8080`,
```
Comment (drxeno02) [inline src/environments/environment.ts:8]: 👍

**Comment by drxeno02** [comment]
@Bithika-Redapple This PR has been approved, will you merge? Lets not keep it sitting. I will merge after a while if there is nothing more to add to this.

**Comment by Bithika-Redapple** [comment]
yes I have already merged this branch with develop/1.0.0

**Comment by drxeno02** [comment]
@Bithika-Redapple Feel free to merge when you are ready.

**Pull request merged**